### PR TITLE
Integration test: Bid Caching / "Immediate" bid (CPM > 0, TTL = 0) 

### PR DESCRIPTION
**Pull request by vguerci**
Also refactor the code a bit to improve code style and testability

### CODE REVIEW

File: CriteoPublisherSdk/Sources/CR_CdbBid.h [hunk 1]
```diff
@@ -74,6 +79,8 @@
 - (BOOL)isExpired;
 - (BOOL)isValid;
 
+- (void)setDefaultTtl;
```
Comment (Amokrane) [inline CriteoPublisherSdk/Sources/CR_CdbBid.h:82]: I suggest adding the time unit to the method name.

File: CriteoPublisherSdk/Sources/CR_CdbBid.m [hunk 2]
```diff
@@ -95,12 +94,8 @@ - (instancetype)initWithDict:(NSDictionary *)slot receivedAt:(NSDate *)receivedA
     NSNumber *height = slot[@"height"];
     NSString *creative = slot[@"creative"];
     NSString *impId = slot[@"impId"];
-    // Hard coding to 15 minutes for now
-    // TODO: move this default to the config
-    NSTimeInterval ttl = (slot && slot[@"ttl"]) ? [slot[@"ttl"] doubleValue] : defaultValue;
-    if ([cpm doubleValue] > 0 && ttl == 0) {
-      ttl = defaultValue;
-    }
+    NSTimeInterval ttl =
```
Comment (Amokrane) [inline CriteoPublisherSdk/Sources/CR_CdbBid.m:97]: Same: adding the time unit to the variable name.

File: CriteoPublisherSdk/Sources/CR_CdbBid.m [hunk 3]
```diff
@@ -215,13 +210,19 @@ - (BOOL)isInSilenceMode {
...
+  return self.isValidCpm && self.isValidNativeAssetsOrUrl;
 }
 
 - (BOOL)isRenewable {
-  BOOL result = !self.isInSilenceMode || self.isExpired;
-  return result;
+  return !self.isInSilenceMode || self.isExpired;
+}
+
+- (BOOL)isImmediate {
```
Comment (Amokrane) [inline CriteoPublisherSdk/Sources/CR_CdbBid.m:220]: nit: could be a good occasion to document each of the `isXXX` methods to educate the reader on the meaning of these different modes.

**Review by Amokrane — CHANGES_REQUESTED** [review]

File: CriteoPublisherSdk/Sources/CR_CdbBid.m (hunk 2 above)
Comment (vguerci) [reply CriteoPublisherSdk/Sources/CR_CdbBid.m:97]: I doubt there is any value naming the variable that way as `NSTimeInterval` type has [_by definition_](https://developer.apple.com/documentation/foundation/nstimeinterval) a second unit.

File: CriteoPublisherSdk/Sources/CR_CdbBid.h (hunk 1 above)
Comment (vguerci) [reply CriteoPublisherSdk/Sources/CR_CdbBid.h:82]: I'm not sure this is something we want to do. This method basically sets the Default TTL into in inner state, from outside caller this is regardless of the underlying unit which is an implementation detail.

File: CriteoPublisherSdk/Sources/CR_CdbBid.m (hunk 3 above)
Comment (vguerci) [reply CriteoPublisherSdk/Sources/CR_CdbBid.m:220]: Agreed but FAR, are you thinking about more documentation than the one I've put?
> YES if a bid is "immediate". i.e. CPM > 0, TTL = 0
Also, is there any reference documentation about all these modes?

File: CriteoPublisherSdk/Sources/CR_CdbBid.m (hunk 2 above)
Comment (Amokrane) [reply CriteoPublisherSdk/Sources/CR_CdbBid.m:97]: Ah ok, well it has some value for someone that doesn't know Objective-C such as myself :) But agreed that it is not necessary in this case.

File: CriteoPublisherSdk/Sources/CR_CdbBid.m (hunk 3 above)
Comment (Amokrane) [reply CriteoPublisherSdk/Sources/CR_CdbBid.m:220]: Yes I was thinking about something more explicit that explains what `immediate` means for example. But it's really nit. There is the PubSDK overview which is pinned on the Slack channel.

**Review by Amokrane — APPROVED** [review]

File: CriteoPublisherSdk/Sources/CR_CdbBid.m (hunk 3 above)
Comment (vguerci) [reply CriteoPublisherSdk/Sources/CR_CdbBid.m:220]: Ok will add some entry in my _personal_ todo list then :)

**Pull request merged**